### PR TITLE
SEO queries revisited

### LINKED ISSUE #91: Query metrics: Chapter 10. SEO
Part | Chapter | Authors | Reviewers | Tracking Issue
-- | -- | -- | -- | --
II. User Experience | 10. SEO | @rachellcostello @ymschaap @AVGP | @clarkeclark @andylimn @voltek62 | #12

# READ ME!
All of the metrics in the table below have been marked as `Able To Query` during the [metrics triage](https://github.com/HTTPArchive/almanac.httparchive.org/issues/33). The analyst assigned to each metric is expected to write the corresponding query and submit a PR to [have it reviewed and added to the repo](https://github.com/HTTPArchive/almanac.httparchive.org/issues/62).

In order to stay on schedule and have the data ready for authors, **please have all metrics reviewed and merged by August 5**.

## Assignments

ID | Metric description | Analyst | Notes
-- | -- | -- | --
10.01 | Structured data rich results eligibility (ratings, search, etc,) | @ymschaap | html -> regex
10.02 | Lang attribute usage and mistakes (lang='en') | @ymschaap | lighthouse -> html-lang-valid, resource: See: https://discuss.httparchive.org/t/what-are-the-invalid-uses-of-the-lang-attribute/1022
10.03 | `<link>` rel="amphtml" (AMP) | @ymschaap | html -> regex
10.04 | `<link>` hreflang="en-us" (localisation usage) | @ymschaap | html -> regex + lighthouse -> hreflang
10.05 | Breakdown of type of structured data served (ld+json, microformatting, schema.org + what `@type`)? | @ymschaap | Custom Query Can we have this data: https://search.google.com/structured-data/testing-tool
10.06 | Indexability - looking at meta tags like `<meta>` noindex, `<link>` canonicals. | @ymschaap | lighthouse -> is-crawlable, lighthouse -> canonical
10.07 | `<meta>` description + `<title>` (presence & length) | @ymschaap | html -> regex
10.08 | Status codes and whether pages are accessible - 200, 3xx, 4xx, 5xx. | @ymschaap | request -> response
10.09 | Content - looking at word count, thin pages, header usage, alt attributes images | @ymschaap | lighthouse ->image-alt,  Custom Query
10.10 | Linking - extract `<a href>` count per page (internal + external) | @ymschaap | Custom Query
10.11 | Linking - fragment URLs (together with SPAs to navigate content) | @ymschaap | we have react/vue as application type + a href Custom Query
10.12 | robots.txt (It is mentioned in Lighthouse, can we parse the content or only confirm its existence? E.g. check if has a sitemap reference - seems it does list the potential issues) | @ymschaap | lighthouse -> robots-txt
10.13 | If the desktop site is responsive/mobile-ready, or a specific mobile site (redirect, UA)? (Can we find if these are different sites?) | @ymschaap | compare mobile vs desktop crawl page -> _final_url + lighthouse -> seo-mobile
10.14 | Descriptive link text usage (available in Lighthouse data) | @ymschaap | lighthouse -> link-text
10.15 | speed metrics (FCP, server response time) would be nice for SEO as well given the recent focus on fast loading sites | @ymschaap | See: https://discuss.httparchive.org/t/measuring-cms-host-ttfb-in-crux/1676/1

## Checklist of metrics to be merged

- [x] 	10.01	Structured data rich results eligibility (ratings, search, etc,)
- [x] 	10.02	Lang attribute usage and mistakes (lang='en')
- [x] 	10.03	`<link>` rel="amphtml" (AMP)
- [x] 	10.04	`<link>` hreflang="en-us" (localisation usage)
- [x] 	10.05	Breakdown of type of structured data served (ld+json, microformatting, schema.org + what `@type`)?
- [x] 	10.06	Indexability - looking at meta tags like `<meta>` noindex, `<link>` canonicals.
- [x] 	10.07	`<meta>` description + `<title>` (presence & length)
- [x] 	10.08	Status codes and whether pages are accessible - 200, 3xx, 4xx, 5xx.
- [x] 	10.09	Content - looking at word count, thin pages, header usage, alt attributes images
- [x] 	10.10	Linking - extract `<a href>` count per page (internal + external)
- [x] 	10.11	Linking - fragment URLs (together with SPAs to navigate content)
- [x] 	10.12	robots.txt (It is mentioned in Lighthouse, can we parse the content or only confirm its existence? E.g. check if has a sitemap reference - seems it does list the potential issues)
- [x] 	10.13	If the desktop site is responsive/mobile-ready, or a specific mobile site (redirect, UA)? (Can we find if these are different sites?)
- [x] 	10.14	Descriptive link text usage (available in Lighthouse data)
- [x] 	10.15	speed metrics (FCP, server response time) would be nice for SEO as well given the recent focus on fast loading sites

### PR DESCRIPTION
**Pull request by rviscomi**
Closes #91 

Results saved to [this sheet](https://docs.google.com/spreadsheets/d/1uARtBWwz9nJOKqKPFinAMbtoDgu5aBtOhsBNmsCoTaA/edit?usp=sharing)

### CODE REVIEW

File: sql/2019/10_SEO/10_01.sql [hunk 1]
```diff
@@ -1,11 +1,8 @@
 #standardSQL
-
-# structured data rich results eligibility
-
+# 10_01: structured data rich results eligibility
 # note: the RegExp options based on: https://developers.google.com/search/docs/guides/search-gallery
 # note: homepage only data
 # note: also see 10.05
-
 CREATE TEMPORARY FUNCTION parseStructuredData(payload STRING)
```
Comment (obto) [inline sql/2019/10_SEO/10_01.sql]: Let's give this a more descriptive name like... hasEligibleType?

File: sql/2019/10_SEO/10_05.sql [hunk 2]
```diff
@@ -1,7 +1,5 @@
 #standardSQL
-
-# structured data by @type
-
+# 10_05: structured data by @type
 CREATE TEMPORARY FUNCTION parseStructuredData(payload STRING)
```
Comment (obto) [inline sql/2019/10_SEO/10_05.sql]: Rename to something like getSchemaTypes?

File: sql/2019/10_SEO/10_07a.sql [hunk 3]
```diff
@@ -1,19 +1,14 @@
 #standardSQL
-
-# <title> and <meta description> present
-
+# 10_07a: <title> and <meta description> present
```
Comment (obto) [inline sql/2019/10_SEO/10_07a.sql]: I think the analysts will also want to know how many sites have both of these present

File: sql/2019/10_SEO/10_07b.sql [hunk 4]
```diff
@@ -1,13 +1,21 @@
 #standardSQL
-
-# <title> length
-
+# 10_07b: <title> length
```
Comment (obto) [inline sql/2019/10_SEO/10_07b.sql]: I've run into a lot of sites which have more than one title tag unfortunately.

Adding this as a custom metric would be very simple, so we could update the query for October's run and see how different the data ends up being?

Comment (rviscomi) [reply sql/2019/10_SEO/10_07b.sql]: This query should only measure the first title, which is what a custom metric would assumedly do too.

Implementing it as a custom metric would also cut it really close to the scheduled launch in early November, considering that we wouldn't have data until the end of October and authors/translators would need to revise the chapter contents.

Comment (obto) [reply sql/2019/10_SEO/10_07b.sql]: Fair enough

**Pull request merged**